From 2fcdf0c58e54230dbfd87e432a3e984ab4579bc8 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Wed, 7 Apr 2010 20:09:34 +0800
Subject: [PATCH 09/22] fix capabilities definition

For "capabilties^{}" lines, replace obj-id with zero-id.

Define capabilities using rules from pack-protocol.txt - ideally, this
should be shifted to protocol-common.txt.

Also, update the sample conversation by removing the SP between NUL and
cap-list.
---
 Documentation/technical/http-protocol.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/Documentation/technical/http-protocol.txt b/Documentation/technical/http-protocol.txt
index e52e604..e7c05d2 100644
--- a/Documentation/technical/http-protocol.txt
+++ b/Documentation/technical/http-protocol.txt
@@ -222,7 +222,7 @@ The request MUST NOT contain additional query parameters.
    S: Cache-Control: no-cache
    S:
    S: ....# service=git-upload-pack
-   S: ....95dcfa3633004da0049d3d0fa03f80589cbcaf31 refs/heads/maint\0 multi_ack
+   S: ....95dcfa3633004da0049d3d0fa03f80589cbcaf31 refs/heads/maint\0multi_ack
    S: ....d049f6c27a2244e12041955e262a404c7faba355 refs/heads/master
    S: ....2cb58b79488a98d2721cea644875a8dd0026b115 refs/tags/v1.0
    S: ....a3c2e2402b99163d1d59756e5f207ae21cccba4c refs/tags/v1.0^{}
@@ -281,14 +281,16 @@ declarations behind a NUL on the first ref.
                      "0000"
   ref_list        =  empty_list / populated_list
 
-  empty_list      =  PKT-LINE(obj-id SP "capabilities^{}" NUL cap_list LF)
+  empty_list      =  PKT-LINE(zero-id SP "capabilities^{}" NUL cap-list LF)
 
   non_empty_list  =  PKT-LINE(obj-id SP name NUL cap_list LF)
                     *ref_record
 
-  cap_list        =  *(SP capability) SP
-  ref_record      =  any_ref / peeled_ref
+  cap-list        =  capability *(SP capability)
+  capability      =  1*(LC_ALPHA / DIGIT / "-" / "_")
+  LC_ALPHA        =  %x61-7A
 
+  ref_record      =  any_ref / peeled_ref
   any_ref         =  PKT-LINE(obj-id SP name LF)
   peeled_ref      =  PKT-LINE(obj-id SP name LF)
                      PKT-LINE(obj-id SP name "^{}" LF
-- 
1.8.4.rc4.527.g303b16c

